File: server/application-server/src/main/java/de/tum/in/www1/hephaestus/teamleaderboard/TeamLeaderboardEntryDTO.java
package de.tum.in.www1.hephaestus.teamleaderboard;

import de.tum.in.www1.hephaestus.gitprovider.pullrequest.PullRequestInfoDTO;
import de.tum.in.www1.hephaestus.gitprovider.team.TeamInfoDTO;
import io.micrometer.common.lang.NonNull;
import java.util.List;

public record TeamLeaderboardEntryDTO(
    @NonNull Integer rank,
    @NonNull Integer score,
    @NonNull TeamInfoDTO team,
    @NonNull List<PullRequestInfoDTO> reviewedPullRequests,
    @NonNull Integer numberOfReviewedPRs,
    @NonNull Integer numberOfApprovals,
    @NonNull Integer numberOfChangeRequests,
    @NonNull Integer numberOfComments,
    @NonNull Integer numberOfUnknowns,
    @NonNull Integer numberOfCodeComments
    ) {}

File: server/application-server/src/main/java/de/tum/in/www1/hephaestus/teamleaderboard/kotlin/TeamLeaderboardEntryDTO.kt
package de.tum.`in`.www1.hephaestus.teamleaderboard.kotlin

import de.tum.`in`.www1.hephaestus.gitprovider.pullrequest.PullRequestInfoDTO
import de.tum.`in`.www1.hephaestus.gitprovider.team.TeamInfoDTO

data class TeamLeaderboardEntryDTO(
    val rank: Int,
    val score: Int,
    val team: TeamInfoDTO,
    val reviewedPullRequests: List<PullRequestInfoDTO>,
    val numberOfReviewedPRs: Int,
    val numberOfApprovals: Int,
    val numberOfChangeRequests: Int,
    val numberOfComments: Int,
    val numberOfUnknowns: Int,
    val numberOfCodeComments: Int,
)